### PR TITLE
Create listings permalink

### PR DESCRIPTION
**Pull request by jessleenyc**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Documentation Update

## Description
- Create slug for each listing
- Update routes
- We'll want to backfill existing listings w/ a slug.

### CODE REVIEW

File: app/models/classified_listing.rb [hunk 1]
```diff
@@ -96,4 +97,8 @@ def validate_tags
   def validate_category
     errors.add(:category, "not a valid category") unless ClassifiedListing.categories_available[category]
   end
+
+  def create_slug
```
Comment (benhalpern) [inline app/models/classified_listing.rb]: The slug also needs to be added to Algolia index

File: app/views/classified_listings/index.html.erb [hunk 2]
```diff
@@ -21,18 +21,22 @@
...
+        only: %i[title processed_html tag_list category id user_id slug],
+        include: {
+          author: { only: %i[username name], methods: %i[username profile_image_90] }
+        },
+      ) %>"
+    data-allcategories="<%= ClassifiedListing.categories_for_display.to_json %>">
     <div class="classified-filters">
     <div class="classified-filters-categories">
-      <a href="/listings" class="<%= "selected" if params[:category].blank? %> data-no-instant="true">all</a><% ClassifiedListing.categories_for_display.each do |cat| %><a href="/listings/<%= cat[:slug] %>" class="<%= "selected" if params[:category] == cat[:slug] %> data-no-instant="true"><%= cat[:name] %></a><% end %><a href='/listings/new' class='classified-create-link'>Create a Listing</a>
+      <a href="/listings" class="<%= "selected" if params[:category].blank? %>" data-no-instant="true">all</a>
```
Comment (benhalpern) [inline app/views/classified_listings/index.html.erb]: This refactor is causing it to "jump". The whitespace is treated differently in the JS vs here so I'm not sure how to make it work with this refactoring. But either way, needs to be sure it won't jump on load.

**Comment by benhalpern** [comment]
Changes look good. Will merge if tests pass

**Comment by jessleenyc** [comment]
@benhalpern i might have pushed an additional changes after you left this comment -- take another just to make sure please

**Comment by benhalpern** [comment]
Cool

File: app/controllers/classified_listings_controller.rb [hunk 3]
```diff
@@ -13,7 +13,9 @@ def index
     set_surrogate_key_header "classified-listings-#{params[:category]}"
   end
 
-  def show; end
+  def show
+    @classified_listing = ClassifiedListing.find_by(category: params[:category], slug: params[:slug])
```
Comment (rhymes) [inline app/controllers/classified_listings_controller.rb]: should this be `find_by!` ? in case the combination category and slug doesn't exist...

**Pull request merged**